### PR TITLE
Add Option/Maybe CE

### LINKED ISSUE #65: Add Option/Maybe CE
Doesn't it make sense to also add Option computation expression to this library?

### PR DESCRIPTION
**Pull request by TheAngryByrd**
This adds Option CE's.  Closes #65.

I wanted to get some feedback so I'm opening it now. I've left this as WIP since I still have to implement other CEs and add documentation.

- [x] sync option
- [x] sync option docs
- [x] async option
- [x] async option docs
- [x] task option
- [x] task option docs
- [x] job option
- [x] task option docs

### CODE REVIEW

File: src/FsToolkit.ErrorHandling/FsToolkit.ErrorHandling.fsproj [hunk 1]
```diff
@@ -34,15 +20,12 @@
...
     <Compile Include="Option.fs" />
+    <Compile Include="OptionCE.fs" />
+    <Compile Include="AsyncOptionCE.fs" />
     <Compile Include="List.fs" />
     <None Include="Script.fsx" />
+    <Content Include="*.fsproj; **/*.fs" />
   </ItemGroup>
-
-  <!-- Add source files to "fable" folder in Nuget package -->
-  <ItemGroup>
```
Comment (TheAngryByrd) [inline src/FsToolkit.ErrorHandling/FsToolkit.ErrorHandling.fsproj]: Grr I think ionide messed this up. Lemme fix it.

**Comment by TheAngryByrd** [comment]
@cmeeren I would love a review from you as well!

**Comment by cmeeren** [comment]
Good job! I don't have the capacity to do a thorough review of the details, but it looks fine at a quick glance, and since you have tests, it should work. :)

**Comment by demystifyfp** [comment]
Thanks @TheAngryByrd for the efforts. It looks good to me. 

Sorry for the delayed response. I was on a month-long vacation.

**Comment by TheAngryByrd** [comment]
Where can I get one of those? 😃

**Comment by demystifyfp** [comment]
@TheAngryByrd You just have to decide :)

**Comment by TheAngryByrd** [comment]
@demystifyfp I'm gonna merge this.  Please make a release whenever you get a chance.

**Comment by demystifyfp** [comment]
Thanks, @TheAngryByrd. 

[It's up](https://www.nuget.org/packages/FsToolkit.ErrorHandling/1.2.6) now!

**Pull request merged**